flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/NettyMessageSerializationTest.java

```diff
@@ -174,6 +197,18 @@ private void testEncodeDecodeBuffer(boolean testReadOnlyBuffer) {
 		assertEquals(expected.backlog, actual.backlog);
 	}
 
+	private Buffer compress(Buffer buffer) {
+		return COMPRESSOR.compressInPlace(buffer);
+	}
+
+	private ByteBuf decompress(ByteBuf buffer) {
+		MemorySegment segment = MemorySegmentFactory.allocateUnpooledSegment(BUFFER_SIZE);
+		Buffer compressedBuffer = new NetworkBuffer(segment, FreeingBufferRecycler.INSTANCE);
+		buffer.readBytes(compressedBuffer.asByteBuf(), buffer.readableBytes());
+		compressedBuffer.setCompressed(true);
+		return DECOMPRESSOR.decompressInPlace(compressedBuffer).asByteBuf();
```

Comment:
All the above processes are actually done in `decompressToInternalBuffer`, why not use that directly?

Reply:
The input is netty's ByteBuf not a Buffer.